x-pack/legacy/plugins/ml/public/components/chart_tooltip/chart_tooltip.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import classNames from 'classnames';
+import React, { useRef, FC } from 'react';
+import { TooltipValue, TooltipValueFormatter } from '@elastic/charts';
+
+// TODO: Below import is temporary, use `react-use` lib instead.
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { useObservable } from '../../../../../../../src/plugins/kibana_react/public/util/use_observable';
+
+import { chartTooltip$, mlChartTooltipService } from './chart_tooltip_service';
+
+const renderHeader = (headerData?: TooltipValue, formatter?: TooltipValueFormatter) => {
+  if (!headerData) {
+    return null;
+  }
+
+  return formatter ? formatter(headerData) : headerData.name;
+};
+
+export const ChartTooltip: FC = () => {
+  const chartTooltipElement = useRef(null);
+
+  mlChartTooltipService.element = chartTooltipElement.current;
+
+  const chartTooltipState = useObservable(chartTooltip$);
+
+  if (chartTooltipState === undefined || !chartTooltipState.isTooltipVisible) {
+    return <div className="mlChartTooltip mlChartTooltip--hidden" ref={chartTooltipElement} />;
+  }
+
+  const { tooltipData, tooltipHeaderFormatter, tooltipPosition } = chartTooltipState;
+
+  return (
+    <div
+      className="mlChartTooltip"
+      style={{ transform: tooltipPosition.transform }}
```

Comment:
The positioning of the tooltip doesn't seem to take into account the position on the screen. It should transform to the left when the hover is towards the right hand edge of the screen:

![image](https://user-images.githubusercontent.com/7405507/66817214-43b7cb80-ef33-11e9-99d9-8f99bb09aa00.png)

Reply:
Fixed in 52d673c.